File: java-springboot-toy/src/main/java/com/ex/javaspringboottoy/config/QueryDslConfig.java
package com.ex.javaspringboottoy.config;

import com.querydsl.jpa.impl.JPAQuery;
import com.querydsl.jpa.impl.JPAQueryFactory;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import javax.persistence.EntityManager;

@RequiredArgsConstructor
@Configuration
public class QueryDslConfig {

    private final EntityManager em;

    @Bean
    public JPAQueryFactory jpaQueryFactory() {
        return new JPAQueryFactory(em);
    }
}

File: spring-jpa/jpa-core/src/main/kotlin/com/study/jpacore/config/QueryDslConfig.kt
package com.study.jpacore.config

import com.querydsl.jpa.impl.JPAQueryFactory
import jakarta.persistence.EntityManager
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class QueryDslConfig {

  @Bean
  fun jpaQueryFactory(entityManager: EntityManager) = JPAQueryFactory(entityManager)
}
